static/js/topic_generator.js

```diff
@@ -80,14 +80,19 @@ exports.wrap_exclude = function (lst, val) {
     if (val === undefined) {
         return exports.list_generator(lst);
     }
-
-    var i = _.indexOf(lst, val);
+    var i;
+    for (i in lst) {
+        if (lst[i].stream === val.stream && lst[i].topic === val.topic) {
+            break;
+        }
+        i = -1;
+    }
+    i = parseInt(i);
     if (i < 0) {
         return exports.list_generator(lst);
     }
-
     var inners = [
-         exports.sub_list_generator(lst, i+1, lst.length),
+         exports.sub_list_generator(lst, i + 1, lst.length),
          exports.sub_list_generator(lst, 0, i),
     ];
```

Comment:
This takes care of the cycling issue. Earlier I had changed the `exports.sub_list_generator(lst, 0, i)` to `exports.sub_list_generator(lst, 0, 0)`